### PR TITLE
typified vehicle stuff

### PR DESCRIPTION
**Pull request by PatrikLundell**
#### Summary
None

#### Purpose of change
Replace untyped coordinate usage with typed usage.

#### Describe the solution
Go through the remainder of vehicle.h, sweep over vehicle.cpp, and go through the rest of the vehicle related .h files to update operation, duplicate ones that still depend on untyped usage, and replace unused untyped operation. This process involves some modification to usages.

#### Describe alternatives you've considered
Try to fix this section of vehicle.cpp:
```
            // TODO: Check if this is correct. part_at takes a vehicle relative position, not a bub one...
            int part_idx = part_at( p.xy().raw() );
```
but decided to leave it as a turd for later clean up. Didn't find any operation that actually took bubble coordinates, so fixing this would require subtracting the vehicle coordinates from the bubble one to actually check the correct location.

#### Testing
Loaded a save, walked up a ramp, jumped into a car, drove through some hay bales, ran over a zombie corpse that had stuff in the inventory, ran over a turkey, and smashed into a stationary vehicle. Didn't see anything odd (the changes should not change any functionality).

#### Additional context
Had a local environment corruption, so it had to be rebuilt. This involved downloading the latest master, creating a branch, and copy the modified files based on a previous master over it to recover the work done. This means that version tags and things like that may be wrong, However, the test was performed after recovery, so at least it's consistent internally.

Also note that typification PRs tend to mess with a lot of places, so there's a significant risk of a merge clash if this PR is merged concurrently with other code PRs.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Could you please remove Respirator-related stuff from the PR?

**Comment by Maleclypse** [comment]
You may have already tried this but have you rebased on top of the current master?

**Comment by PatrikLundell** [comment]
I tried to rebase, and the results compile. However, when trying to push the results it was refused because the local stuff was considered older than the remote stuff (somehow). I ended up using --force to force the stuff through, which may be incorrect, as I don't really know what I'm doing.

File: src/bionics.cpp [hunk 1]
```diff
@@ -1693,13 +1693,10 @@ void Character::process_bionic( bionic &bio )
             mod_power_level( -trigger_cost );
         }
     } else if( bio.id == bio_gills ) {
-        const units::energy trigger_cost = bio.info().power_trigger / 8;
-        if( has_effect( effect_asthma ) && get_power_level() >= trigger_cost ) {
+        if( has_effect( effect_asthma ) ) {
```
Comment (Night-Pryanik) [inline src/bionics.cpp:1696]:
```suggestion
        const units::energy trigger_cost = bio.info().power_trigger / 8;
        if( has_effect( effect_asthma ) && get_power_level() >= trigger_cost ) {
```

File: src/bionics.cpp [hunk 2]
```diff
@@ -1693,13 +1693,10 @@ void Character::process_bionic( bionic &bio )
...
             add_msg_if_player( m_good,
-                               _( "Your %s activates and you feel your throat open up and air filling your lungs!" ),
-                               bio.info().name );
+                               _( "You feel your throat open up and air filling your lungs!" ) );
```
Comment (Night-Pryanik) [inline src/bionics.cpp:1698]:
```suggestion
                               _( "Your %s activates and you feel your throat open up and air filling your lungs!" ),
                               bio.info().name );
```

File: src/bionics.cpp [hunk 3]
```diff
@@ -1693,13 +1693,10 @@ void Character::process_bionic( bionic &bio )
...
             remove_effect( effect_asthma );
```
Comment (Night-Pryanik) [inline src/bionics.cpp:1699]:
```suggestion
            remove_effect( effect_asthma );
            mod_power_level( -trigger_cost );
```

**Review by Night-Pryanik — COMMENTED** [review]
How about that?

File: src/bionics.cpp (hunk 3 above)
Comment (PatrikLundell) [reply src/bionics.cpp:1701]: How would I know? I haven't changed it and I don't know where it comes from. The tools haven't flagged it as needing conflict resolution, so I haven't had any reason to change it back. I have not opinion about changes to that code in either direction, so I don't have any problem accepting suggested changes.

**Pull request merged**